lms/djangoapps/instructor/enrollment.py

```diff
@@ -317,7 +317,11 @@ def _fire_score_changed_for_block(course_id, student, block, module_state_key):
             field_data_cache=cache,
             course_key=course_id
         )
-        points_earned, points_possible = weighted_score(0, module.max_score(), getattr(module, 'weight', None))
+        max_score = module.max_score()
+        if max_score is None:
+            points_earned, points_possible = 0, 0
+        else:
+            points_earned, points_possible = weighted_score(0, max_score, getattr(module, 'weight', None))
     else:
         points_earned, points_possible = 0, 0
```

Comment:
And the code to send the SCORE_CHANGED signal is never hit when we try to delete a student's state for a block with no score. See https://github.com/edx/edx-platform/pull/13539#discussion_r81147437 

The check for `has_score` was written with the assumption that `has_score == True` would be 1-to-1 with the block defining a max score. Now we know that's not true!